lib/sequel/adapters/shared/mssql.rb

```diff
@@ -476,6 +476,21 @@ def set_mssql_unicode_strings
         @mssql_unicode_strings = typecast_value_boolean(@opts.fetch(:mssql_unicode_strings, true))
       end
       
+
+      # Specify unicode strings by default for Strings.
+      # If a size isn't present, assumes a size of 255.
+      # If the :fixed option is used, uses the nchar type.
+      # If the :text option is used, MSSQL prefers nvarchar(max) as text and ntext are depricated.
+      def type_literal_generic_string(column)
+        if column[:text]
+          "nvarchar(max)"
```

Comment:
As `TEXT` is [deprecated](https://docs.microsoft.com/en-us/sql/t-sql/data-types/ntext-text-and-image-transact-sql?view=sql-server-ver15) and often throws errors when you try to use it, it generally makes sense to set this to be `NVARCHAR(MAX)` as recommended by MS. However, do we need to check server version for this? How far back can this recommendation extend? Jeremy often wants to preserve as much backwards compat as possible, and MSSQL is notorious for running ancient versions.